### PR TITLE
Bump to GraalVM 22.2.0

### PR DESCRIPTION
**Pull request by gastaldi**
Requires https://github.com/quarkusio/quarkus-images/pull/201 to be merged first

### CODE REVIEW

**Review by gsmet — CHANGES_REQUESTED** [review]
I think some of the changes are missing. See https://github.com/quarkusio/quarkus/pull/25366/files for reference.

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java [hunk 1]
```diff
@@ -16,8 +16,8 @@
 @ConfigRoot(phase = ConfigPhase.BUILD_TIME)
 public class NativeConfig {
 
-    public static final String DEFAULT_GRAALVM_BUILDER_IMAGE = "quay.io/quarkus/ubi-quarkus-native-image:22.1-java17";
-    public static final String DEFAULT_MANDREL_BUILDER_IMAGE = "quay.io/quarkus/ubi-quarkus-mandrel:22.1-java17";
+    public static final String DEFAULT_GRAALVM_BUILDER_IMAGE = "quay.io/quarkus/ubi-quarkus-native-image:22.2-java17";
+    public static final String DEFAULT_MANDREL_BUILDER_IMAGE = "quay.io/quarkus/ubi-quarkus-mandrel:22.2-java17";
```
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java:20]: Mandrel 22.2 has not been released yet and is also not covered by https://github.com/quarkusio/quarkus-images/pull/201

**Review by zakkak — CHANGES_REQUESTED** [review]

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java [hunk 2]
```diff
@@ -82,7 +82,7 @@ public class NativeImageBuildStep {
     void addExportsToNativeImage(BuildProducer<JPMSExportBuildItem> exports) {
         // Needed by io.quarkus.runtime.ResourceHelper.registerResources
         exports.produce(new JPMSExportBuildItem("org.graalvm.nativeimage.builder", "com.oracle.svm.core.jdk",
-                GraalVM.Version.VERSION_22_1_0));
+                GraalVM.Version.CURRENT));
```
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java:85]: This should not change. It says that the export should be done for GraalVM >= 22.1

File: core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java [hunk 3]
```diff
@@ -127,11 +127,11 @@ void addExportsToNativeImage(BuildProducer<JPMSExportBuildItem> features,
             List<NativeImageResourcePatternsBuildItem> resourcePatterns) {
         // required in order to access org.graalvm.nativeimage.impl.RuntimeSerializationSupport and org.graalvm.nativeimage.impl.ConfigurationCondition
         features.produce(
-                new JPMSExportBuildItem("org.graalvm.sdk", "org.graalvm.nativeimage.impl", GraalVM.Version.VERSION_22_1_0));
+                new JPMSExportBuildItem("org.graalvm.sdk", "org.graalvm.nativeimage.impl", GraalVM.Version.CURRENT));
```
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java:130]: This should not change. It says that the export should be done for GraalVM >= 22.1

File: core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java [hunk 4]
```diff
@@ -127,11 +127,11 @@ void addExportsToNativeImage(BuildProducer<JPMSExportBuildItem> features,
...
         // required in order to access com.oracle.svm.core.jni.JNIRuntimeAccess
         if (jniRuntimeAccessibleClasses != null && !jniRuntimeAccessibleClasses.isEmpty()) {
             features.produce(new JPMSExportBuildItem("org.graalvm.nativeimage.builder", "com.oracle.svm.core.jni",
-                    GraalVM.Version.VERSION_22_1_0));
+                    GraalVM.Version.CURRENT));
```
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java:134]: This should not change. It says that the export should be done for GraalVM >= 22.1

File: core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java [hunk 5]
```diff
@@ -260,7 +260,7 @@ public void write(String s, byte[] bytes) {
         if (!proxies.isEmpty()) {
             // Needed to access DYNAMIC_PROXY_REGISTRY
             exports.produce(new JPMSExportBuildItem("org.graalvm.nativeimage.builder", "com.oracle.svm.core.jdk.proxy",
-                    GraalVM.Version.VERSION_22_1_0));
+                    GraalVM.Version.CURRENT));
```
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java:263]: This should not change. It says that the export should be done for GraalVM >= 22.1

File: core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java [hunk 6]
```diff
@@ -284,7 +284,7 @@ public void write(String s, byte[] bytes) {
         if (!resourcePatterns.isEmpty()) {
             // Needed to access LOOKUP_METHOD
             exports.produce(new JPMSExportBuildItem("org.graalvm.nativeimage.base", "com.oracle.svm.util",
-                    GraalVM.Version.VERSION_22_1_0));
+                    GraalVM.Version.CURRENT));
```
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java:287]: This should not change. It says that the export should be done for GraalVM >= 22.1

File: core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java [hunk 7]
```diff
@@ -346,7 +346,7 @@ public void write(String s, byte[] bytes) {
             // Needed to access LOCALIZATION_FEATURE
             exports.produce(
                     new JPMSExportBuildItem("org.graalvm.nativeimage.builder", "com.oracle.svm.core.jdk.localization",
-                            GraalVM.Version.VERSION_22_1_0));
+                            GraalVM.Version.CURRENT));
```
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java:349]: This should not change. It says that the export should be done for GraalVM >= 22.1

File: core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java [hunk 8]
```diff
@@ -508,7 +508,7 @@ public void write(String s, byte[] bytes) {
                 if (registerSerializationMethod == null) {
                     // Needed by createRegisterSerializationForClassMethod to access LOOKUP_METHOD
                     exports.produce(new JPMSExportBuildItem("org.graalvm.nativeimage.base", "com.oracle.svm.util",
-                            GraalVM.Version.VERSION_22_1_0));
+                            GraalVM.Version.CURRENT));
```
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java:511]: This should not change. It says that the export should be done for GraalVM >= 22.1

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java [hunk 9]
```diff
@@ -26,10 +26,10 @@ public static final class Version implements Comparable<Version> {
         static final Version VERSION_21_3_2 = new Version("GraalVM 21.3.2", "21.3.2", Distribution.ORACLE);
         static final Version VERSION_22_0_0_2 = new Version("GraalVM 22.0.0.2", "22.0.0.2", Distribution.ORACLE);
 
-        public static final Version VERSION_22_1_0 = new Version("GraalVM 22.1.0", "22.1.0", Distribution.ORACLE);
```
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java:29]: We need to keep this around as well (see my later comments)

**Review by zakkak — DISMISSED** [review]

**Comment by gastaldi** [comment]
Marking as a draft until the build in https://github.com/quarkusio/quarkus-images is fixed

**Comment by gastaldi** [comment]
Yay! Green CI 🎆 🎉 🎉 🎉

**Review by gsmet — APPROVED** [review]
Looks good to me but let's wait for @zakkak before merging.

**Review by galderz — APPROVED** [review]

**Comment by gastaldi** [comment]
@gsmet Since @zakkak is away all of august and @galderz approved it, maybe we can merge this?

**Pull request merged**